source/extensions/stat_sinks/hystrix/hystrix.h

```diff
@@ -0,0 +1,144 @@
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "envoy/server/admin.h"
+#include "envoy/server/instance.h"
+#include "envoy/stats/stats.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace StatSinks {
+
+typedef std::vector<uint64_t> RollingWindow;
+typedef std::map<const std::string, RollingWindow> RollingStatsMap;
+
+class HystrixStatCache : public Logger::Loggable<Logger::Id::hystrix> {
+public:
+  HystrixStatCache()
+      : current_index_(DEFAULT_NUM_OF_BUCKETS), window_size_(DEFAULT_NUM_OF_BUCKETS + 1){};
+
+  HystrixStatCache(uint64_t num_of_buckets)
```

Comment:
It is confusing.. The statsCache is a *rolling window*. If for example, we have flush interval of 1 ms, and the `number_of_buckets` is 5, it means that we send every 1 ms the statistics for the last 5 ms. The values stored in it are the counters values, which are constantly increasing. So the result we want to return every `endFlush()` is the value of current counter, minus the counter *before* the rolling window started. That's why we keep num_of_buckets+1 values in the rolling window. Does it make sense?

Reply:
Yep, that makes sense.  Thanks for the explanation.  On another note, are 1 or 0 acceptable inputs to this function?  If not, could we add some input validation (either where you read from the config or here)?

Reply:
I assume '0' means unset value, in this case it will be set to default value (10) . '1' should be legitimate value, but it is good to make sure it works. I already did that but config.cc I see here is not the updated one (the github scalability issue?)